src/DeVinci_frontend/components/SidebarHeader.svelte

```diff
@@ -8,11 +8,10 @@
   import ChatHistory from "./ChatHistory.svelte";
 
   import devincilogo from "/devinci-logo.svg";
-
-  import { userHasDownloadedModel } from "../helpers/localStorage";
+  import { downloadedModels } from '../helpers/modelStore';
 
   // Reactive statement to check if the user has already downloaded at least one AI model
-  $: userHasDownloadedAtLeastOneModel = userHasDownloadedModel();
+  $: userHasDownloadedAtLeastOneModel = $downloadedModels.length > 0;
```

Comment:
best to use the existing function userHasDownloadedModel as it implements this already

Reply:
the issue with the previous implementation was that they were not responsive, meaning, after the very first model was downloaded, the chat button was still disabled and info panels on the sidebar + sidebar footer with current model name did not load up. it was only working after refreshing the page.

can we have the current instant responsiveness with the userHasDownloadedModel function?

Reply:
I see, that makes sense. I took a look and it doesn't seem like we can use the existing function with the responsive statement, Let's do the following then;
move downloadedModels to store.ts (as e.g. selectedAiModelId), such that it can be imported from there (and used like currently, e.g. $downloadedModels.length > 0;)
move addDownloadedModel to localStorage.js and integrate it into the existing function there (setLocalFlag), i.e. update downloadedModels and set the local variables in the same place